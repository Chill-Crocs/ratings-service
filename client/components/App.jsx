import React, { Component } from 'react';
import axios from 'axios';

import ReviewList from './ReviewList'

class App extends Component {
  constructor(props) {
    super(props);

    this.state = {
      reviews: [],
      start: 0,
      end: 4
    }
    this.getReviews = this.getReviews.bind(this);
    this.getRandomItem = this.getRandomItem.bind(this);
  }

  componentDidMount() {
    this.getReviews();
  }

  getReviews() {
    axios.get('/api/reviews')
      .then((response) => {
        const items = response.data;
        const reviews = this.getRandomItem(items);
        this.setState({
          reviews: reviews.shopReviews,
          average: reviews.average,
          total: reviews.total
        })
      })
  }

  getRandomItem(shopReviewArray) {
    return shopReviewArray[
      Math.floor(Math.random() * shopReviewArray.length)
    ];
  }

  render() {
    let fourReviews = this.state.reviews.slice(this.state.start, this.state.end)
    return (
      <div>
        <span>{ this.state.reviews.length }</span> Reviews
        <ReviewList reviews={ fourReviews } />
        <div>
          <button>1</button>
          <button>2</button>
          <button>3</button>
          <button>4</button>
        </div>
      </div>
    )
  }
}

export default App;