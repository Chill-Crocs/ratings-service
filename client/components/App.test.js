import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16'
import toJSON from 'enzyme-to-json';
import React from 'react';

import App from './App';

describe('App testing...', () => {
  test('matches snapshot', () => {
    const wrapper = shallow(<App />);

    expect(toJSON(wrapper)).toMatchSnapshot();
  })
})

