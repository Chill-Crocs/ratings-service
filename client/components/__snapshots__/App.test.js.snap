// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App testing... matches snapshot 1`] = `
<div>
  <span>
    0
  </span>
   Reviews
  <ReviewList
    reviews={Array []}
  />
  <div>
    <button>
      1
    </button>
    <button>
      2
    </button>
    <button>
      3
    </button>
    <button>
      4
    </button>
  </div>
</div>
`;
